includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -394,6 +394,28 @@ private function adjust_and_replace_nodes_in_array_map( $node_lists ) {
 		}
 	}
 
+	/**
+	 * Add 'srcset' attribute to image node if lightbox is enabled for that.
+	 *
+	 * @param Element $node Image node.
+	 *
+	 * @return void
+	 */
+	private function set_srcset_for_lightbox_images( Element $node ) {
+
+		if (
+			! $node->hasAttribute( Attribute::SRCSET )
+			&&
+			$node->hasAttribute( 'data-amp-lightbox' )
```

Comment:
Sometimes the `data-amp-lightbox` attribute can be on an ancestor node:

https://github.com/ampproject/amp-wp/blob/49a8eef34942d9c4429bf61be3c949d986fcc7bf/includes/embeds/class-amp-gallery-embed-handler.php#L101-L111

https://github.com/ampproject/amp-wp/blob/a41c2f682caccd6bd67e364dffdfdb1547c07e65/includes/sanitizers/class-amp-gallery-block-sanitizer.php#L80-L85

https://github.com/ampproject/amp-wp/blob/1916a9129c449d13d0072c30a37bac61753ac202/includes/sanitizers/class-amp-img-sanitizer.php#L540-L541

Or will that have always been copied to the `amp-img` at this point?

Reply:
At this point it's always copied to child img or amp-img because of this.


https://github.com/ampproject/amp-wp/blob/2709f5f63f74efc893248e904b10051948ba9c68/includes/embeds/class-amp-core-block-handler.php#L283-L308

Reply:
What about a `gallery` shortcode?

Reply:
Ohh. Right, `gallery` shortcode does add `data-amp-lightbox` to parent.

However, I have remove the condition of checking `data-amp-lightbox` attribute in [this](https://github.com/ampproject/amp-wp/pull/7061/commits/d66a48413179c2aef961e08d53eb8c24456d2722) commit.